### PR TITLE
add-gha-to-run-integration-tests

### PR DESCRIPTION
**Pull request by ykim-1**
add option to makefile to run all the tests including unit and integration fix test_domain_records.py failures due to target variable purge old bats tests

## 📝 Description

- create new yml file for running integration tests with pytest
- add option to makefile to run all the tests including unit and integration
- fix test_domain_records.py failures due to target variable
- purge old bats tests

## ✔️ How to Test

**What are the steps to reproduce the issue or verify the changes?**

**How do I run the relevant unit/integration tests?**

## 📷 Preview

**If applicable, include a screenshot or code snippet of this change. Otherwise, please remove this section.**

### CODE REVIEW

File: .github/workflows/e2e-suite.yml
Comment (zliang-akamai) [inline .github/workflows/e2e-suite.yml:18]:
```suggestion
        uses: actions/setup-python@v4
```
For official actions published by GitHub, let's pin version number instead of hash

**Comment by zliang-akamai** [comment]
Can you also pin checkout action to be `v3`?

`        uses: actions/checkout@v3` at line 12.

File: tests/integration/domains/test_domain_records.py
Comment (zliang-akamai) [inline tests/integration/domains/test_domain_records.py:42]: Having issue with these test cases?

File: .github/workflows/e2e-suite.yml
Comment (lgarber-akamai) [inline .github/workflows/e2e-suite.yml:5]: For security purposes, this workflow should probably only run on `push` rather than on `push` and `pull_request`. 

To address this, we have a [workflow](https://github.com/linode/linode-cli/blob/dev/.github/workflows/e2e-suite-pr-command.yml) to manually trigger E2E tests when a maintainer uses the `/acctest` command. This is necessary as an external contributor could potentially push up malicious code to grab our test account token, etc. without maintainer approval. Fortunately, GitHub Actions only exposes secrets to workflows that explicitly use them, so our other workflows shouldn't be susceptible to this.

Comment (lgarber-akamai) [inline .github/workflows/e2e-suite.yml:5]: For security purposes, this workflow should probably only run on `push` rather than on `push` and `pull_request`. 

To address this, we have a [workflow](https://github.com/linode/linode-cli/blob/dev/.github/workflows/e2e-suite-pr-command.yml) to manually trigger E2E tests when a maintainer uses the `/acctest` command. This is necessary as an external contributor could potentially push up malicious code to grab our test account token, etc. without maintainer approval. Fortunately, GitHub Actions only exposes secrets to workflows that explicitly use them, so our other workflows shouldn't be susceptible to this.

**Review by lgarber-akamai — CHANGES_REQUESTED** [review]
Great work! Just one small change

File: tests/integration/domains/test_domain_records.py
Comment (ykim-1) [reply tests/integration/domains/test_domain_records.py:42]: Yeah these were failing - https://linode.slack.com/archives/GPF2AVATC/p1679678683761609

**Comment by lgarber-akamai** [comment]
/acctest sha=a14c7e43adb4d80aef1b9e16b48eb174e988e44c tests=domains

**Review by lgarber-akamai — APPROVED** [review]

**Pull request merged**
